src/main.cpp

```diff
@@ -740,6 +740,25 @@ bool CheckTransaction(const CTransaction& tx, CValidationState &state)
     return true;
 }
 
+void LimitMempoolSize(CTxMemPool& pool, size_t limit, unsigned long age) {
+    int expired = pool.Expire(GetTime() - age);
+    if (expired != 0)
+        LogPrint("mempool", "Expired %i transactions from the memory pool\n", expired);
+
+    size_t mempoolUsage = pool.DynamicMemoryUsage();
+    size_t coinsUsage = pcoinsTip->DynamicMemoryUsage();
+
+    while (mempoolUsage > limit || coinsUsage + mempoolUsage > nCoinCacheUsage + limit) {
```

Comment:
Yeah, I think that loop and the link with coincache size should go away. The coincache size shouldn't exceed its own limit, and the mempool shouldn't exceed its own limit. We may want to move to a model where the size for both is set using a single limit, but that will require other changes elsewhere too.